pkg/image/image_test.go

```diff
@@ -72,3 +72,154 @@ func TestEdgeSet(t *testing.T) {
+		version         string
+	}
+	tests := []struct {
+		name string
+		args args
+		want Set
+	}{
+		{
+			name: "repo nil, ver not nil",
+			args: args{
+				imageRepository: "",
+				version:         "v1.12.0",
+			},
+			want: Set{
+				CloudAdmission:         "kubeedge/admission:v1.12.0",
+				CloudCloudcore:         "kubeedge/cloudcore:v1.12.0",
+				CloudIptablesManager:   "kubeedge/iptables-manager:v1.12.0",
+				CloudControllerManager: "kubeedge/controller-manager:v1.12.0",
+			},
+		},
+		{
+			name: "repo nil, ver nil",
+			args: args{
+				imageRepository: "",
+				version:         "",
+			},
+			want: Set{
+				CloudAdmission:         "kubeedge/admission",
+				CloudCloudcore:         "kubeedge/cloudcore",
+				CloudIptablesManager:   "kubeedge/iptables-manager",
+				CloudControllerManager: "kubeedge/controller-manager",
+			},
+		},
+		{
+			name: "repo not nil, ver not nil",
+			args: args{
+				imageRepository: "kubeedge-test",
+				version:         "v1.12.0",
+			},
+			want: Set{
+				CloudAdmission:         "kubeedge-test/admission:v1.12.0",
+				CloudCloudcore:         "kubeedge-test/cloudcore:v1.12.0",
+				CloudIptablesManager:   "kubeedge-test/iptables-manager:v1.12.0",
+				CloudControllerManager: "kubeedge-test/controller-manager:v1.12.0",
+			},
+		},
+		{
+			name: "repo not nil, ver nil",
+			args: args{
+				imageRepository: "kubeedge-test",
+				version:         "",
+			},
+			want: Set{
+				CloudAdmission:         "kubeedge-test/admission",
+				CloudCloudcore:         "kubeedge-test/cloudcore",
+				CloudIptablesManager:   "kubeedge-test/iptables-manager",
+				CloudControllerManager: "kubeedge-test/controller-manager",
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := CloudSet(tt.args.imageRepository, tt.args.version); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("CloudSet() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSet_Get(t *testing.T) {
+	tests := []struct {
+		name string
+		s    Set
+		args string
+		want string
+	}{
+		{
+			name: "get cloudcore image",
+			s:    Set{"cloudcore": "kubeedge-test/cloudcore:1.12.0"},
+			args: "cloudcore",
+			want: "kubeedge-test/cloudcore:1.12.0",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.s.Get(tt.args); got != tt.want {
+				t.Errorf("Set.Get() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSet_List(t *testing.T) {
+	tests := []struct {
+		name string
+		s    Set
+		want []string
+	}{
+		{
+			name: "test list",
+			s:    Set{"cloudcore": "kubeedge-test/cloudcore:v1.12.0", "admission": "kubeedge-test/admission:v1.12.0", "controller-manager": "kubeedge-test/controller-manager:v1.12.0", "iptables-manager": "kubeedge-test/iptables-manager:v1.12.0"},
+			want: []string{"kubeedge-test/cloudcore:v1.12.0", "kubeedge-test/admission:v1.12.0", "kubeedge-test/controller-manager:v1.12.0", "kubeedge-test/iptables-manager:v1.12.0"},
```

Comment:
https://pkg.go.dev/reflect#DeepEqual
Array values are deeply equal when their corresponding elements are deeply equal.

ref: https://go.dev/blog/maps
When iterating over a map with a range loop, the iteration order is not specified and is not guaranteed to be the same from one iteration to the next. If you require a stable iteration order you must maintain a separate data structure that specifies that order. This example uses a separate sorted slice of keys to print a map[int]string in key order:

So this test is not right?
@my-git9

Reply:
I'm sorry for that